magenta-lib/src/main/scala/magenta/tasks/tasks.scala

```diff
@@ -274,22 +275,30 @@ trait PollingCheck {
   ): Unit = {
     val expiry = Instant.now().plus(duration)
 
-    def checkAttempt(currentAttempt: Int): Unit = if (!theCheck) {
-      if (stopFlag)
-        reporter.info("Abandoning remaining checks as stop flag has been set")
-      else {
-        val remainingTime = between(Instant.now(), expiry)
-        if (remainingTime.isNegative)
-          reporter.fail(
-            s"Check failed to pass within $duration milliseconds (tried $currentAttempt times) - aborting"
-          )
+    def checkAttempt(currentAttempt: Int): Unit = {
+      val checkResult =
+        Try(theCheck).recover { case NonFatal(e) =>
+          reporter.info(e.getMessage)
+          false
+        }.get // '.get' allows a fatal exception, like an OutOfMemoryError, to kill the process, which is what we want!
```

Comment:
@jacobwinch & I decided that as polling checks inevitably involve network calls, it makes sense to put an exception-catching guard around the invocation of `theCheck` - if a non-fatal exception is received, we probably _don't_ want to kill the deploy, just attempt the check again after the next `sleepyTime`.